autonomi/src/client/archive.rs

```diff
@@ -21,11 +25,40 @@ use xor_name::XorName;
 /// The address of an archive on the network. Points to an [`Archive`].
 pub type ArchiveAddr = XorName;
 
+use thiserror::Error;
+
+#[derive(Error, Debug, PartialEq, Eq)]
+pub enum ArchiveError {
```

Comment:
```suggestion
pub enum RenameError {
```

If we also add a `remove` method later, we might rename this into something else. But let's not use `ArchiveError`, as it's generic and we'll probably make the mistake of piling all our error variants in here.